### PR TITLE
Worked around incorrect warning from nvcc concering unreached return

### PR DESCRIPTION
**Pull request by mcfadden8**

### CODE REVIEW

File: src/umpire/resource/MemoryResourceTypes.hpp [hunk 1]
```diff
@@ -42,6 +42,9 @@ inline std::string resource_to_string(MemoryResourceType type)
       return "FILE";
     default:
       UMPIRE_ERROR("Unkown resource type: " << type);
+#if defined(__CUDACC__)
```
Comment (davidbeckingsale) [inline src/umpire/resource/MemoryResourceTypes.hpp:45]: Can you add a comment about why this is here.

**Comment by mcfadden8** [comment]
@davidbeckingsale, If it is okay with you, I will also be checking in the release artifacts (version number changes, etc. into this PR).

**Comment by mcfadden8** [comment]
LGTM

File: src/umpire/resource/MemoryResourceTypes.hpp [hunk 2]
```diff
@@ -55,6 +58,9 @@ inline MemoryResourceType string_to_resource(const std::string& resource)
   else if (resource == "FILE") return MemoryResourceType::File;
   else {
     UMPIRE_ERROR("Unkown resource name: " << resource);
+#if defined(__CUDACC__)
```
Comment (davidbeckingsale) [inline src/umpire/resource/MemoryResourceTypes.hpp:61]: Add a comment explaining this.

**Review by davidbeckingsale — CHANGES_REQUESTED** [review]
Quick suggestion. Also, you need to update all the version number etc.

**Comment by mcfadden8** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**